src/theme/sw.js

```diff
@@ -0,0 +1,45 @@
+importScripts(
+  "https://storage.googleapis.com/workbox-cdn/releases/4.3.1/workbox-sw.js"
+);
+
+// https://developers.google.com/web/fundamentals/instant-and-offline/offline-cookbook/#stale-while-revalidate
+// TLDR: If there's a cached version available, use it, but fetch an update for next time.
+const staleWhileRevalidate = new workbox.strategies.StaleWhileRevalidate();
+
+// Remote fonts and JavaScript libraries
+workbox.routing.registerRoute(
+  new RegExp("https://fonts.googleapis.com/css"),
+  staleWhileRevalidate
+);
+workbox.routing.registerRoute(
+  new RegExp("https://fonts.gstatic.com"),
+  staleWhileRevalidate
+);
+workbox.routing.registerRoute(
+  new RegExp("https://maxcdn.bootstrapcdn.com/font-awesome"),
+  staleWhileRevalidate
+);
+workbox.routing.registerRoute(
+  new RegExp("https://cdnjs.cloudflare.com/ajax/libs/mathjax"),
+  staleWhileRevalidate
+);
+workbox.routing.registerRoute(
+  new RegExp("https://cdn.jsdelivr.net/clipboard.js"),
+  staleWhileRevalidate
+);
+
+// Local resources
+workbox.routing.registerRoute(
+  new RegExp(".woff2?|.ttf|.css|.js|.json|.png|.svg"),
```

Comment:
Maybe a more robust regex?
And all registered routes above may need some modifications and escaping such as `^`.

```suggestion
  /\.(woff2?|ttf|css|js|json|png|svg)$/
```